Classes/common/CDTFetchChanges.m

```diff
@@ -91,7 +100,10 @@ - (SequenceNumber)notifyChanges:(TD_RevisionList *)changes
     }
```

Comment:
This comment is about line [96](https://github.com/cloudant/CDTDatastore/pull/125/files#diff-bac15dfde454859cd8e3a8dc04fcf29fR96) but I can not make the comment right below it.

Let imagine this case:
1. The operation is not cancelled and pass the first validation(line [52](https://github.com/cloudant/CDTDatastore/pull/125/files#diff-bac15dfde454859cd8e3a8dc04fcf29fR52))
2. It gets all the changes (line [56](https://github.com/cloudant/CDTDatastore/pull/125/files#diff-bac15dfde454859cd8e3a8dc04fcf29fR56)). To do that, it reads from **disk** through the **serial queue** managed by the FMDatabase, therefore it takes a while.
3. The operation is cancelled
4. In line [96](https://github.com/cloudant/CDTDatastore/pull/125/files#diff-bac15dfde454859cd8e3a8dc04fcf29fR96), we get the documents with the specified IDs. As before, to do that we have to read from **disk** through the same **serial queue**, this is slow and costly.
5. Finally, in line [103](https://github.com/cloudant/CDTDatastore/pull/125/files#diff-bac15dfde454859cd8e3a8dc04fcf29fR103), we check if the NSOperation is cancelled.

I think it would be a good idea to check before and after if the NSOperation is cancelled, for example:

``` objc
    if (self.cancelled) {
        return lastSequence;
    }

    NSArray *docs = [_datastore getDocumentsWithIds:[changes allDocIDs]];

    NSMutableDictionary *updatedRevisions = [NSMutableDictionary dictionary];
    for (CDTDocumentRevision *rev in docs) {
        if (self.cancelled) {
            return lastSequence;
        }

        if (rev != nil && !rev.deleted) {
            updatedRevisions[rev.docId] = rev;
        }
    }
```

Or it might be clearer to add this validation right after line [56](https://github.com/cloudant/CDTDatastore/pull/125/files#diff-bac15dfde454859cd8e3a8dc04fcf29fR56). However I recommend to keep the validation inside the loop that iterates through the changed docs (we can not know how long it is going to take):

``` objc
    for (CDTDocumentRevision *rev in docs) {
        if (self.cancelled) {
            return lastSequence;
        }
        ...
    }
```